commands/scale_cluster.go

```diff
@@ -38,108 +38,277 @@ Examples:
+		confirmed := askForConfirmation(fmt.Sprintf("The cluster is autoscaling between %d and %d worker nodes, with %d worker nodes currently desired.\nDo you want to pin the number of worker nodes to %d?", minBefore, maxBefore, desiredCapacity, args.workersMin))
+		if !confirmed {
+			return microerror.Mask(commandAbortedError)
+		}
+	}
+	if maxBefore == minBefore && args.workersMax != args.workersMin {
+		confirmed := askForConfirmation(fmt.Sprintf("The cluster size is currently pinned to %d.\nDo you want to change the limits to be min=%d, max=%d?", minBefore, args.workersMin, args.workersMax))
+		if !confirmed {
+			return microerror.Mask(commandAbortedError)
+		}
+	}
+	if maxBefore != minBefore && args.workersMax != args.workersMin {
+		confirmed := askForConfirmation(fmt.Sprintf("The cluster is autoscaling between %d and %d worker nodes, with %d worker nodes currently up.\nDo you want to change the limits to be min=%d, max=%d?", minBefore, maxBefore, desiredCapacity, args.workersMin, args.workersMax))
+		if !confirmed {
+			return microerror.Mask(commandAbortedError)
+		}
+	}
+
+	return nil
+
 }
 
-func defaultScaleClusterArguments() scaleClusterArguments {
+// defaultScaleClusterArguments defaults arguments supplied by the users.
+func defaultScaleClusterArguments(cmd *cobra.Command, clusterId string, maxBefore int64, minBefore int64) scaleClusterArguments {
 	endpoint := config.Config.ChooseEndpoint(cmdAPIEndpoint)
 	token := config.Config.ChooseToken(endpoint, cmdToken)
 	scheme := config.Config.ChooseScheme(endpoint, cmdToken)
 
-	return scaleClusterArguments{
-		apiEndpoint:         endpoint,
-		authToken:           token,
-		scheme:              scheme,
-		clusterID:           cmdClusterID,
-		numWorkersDesired:   cmdNumWorkers,
-		workerNumCPUs:       cmdWorkerNumCPUs,
-		workerMemorySizeGB:  cmdWorkerMemorySizeGB,
-		workerStorageSizeGB: cmdWorkerStorageSizeGB,
-		oppressConfirmation: cmdForce,
-		verbose:             cmdVerbose,
+	scaleArgs := scaleClusterArguments{
+		apiEndpoint:           endpoint,
+		authToken:             token,
+		clusterID:             clusterId,
+		numWorkersDesired:     cmdNumWorkers,
+		oppressConfirmation:   cmdForce,
+		scheme:                scheme,
+		verbose:               cmdVerbose,
+		workersMax:            cmdWorkersMax,
+		workersMin:            cmdWorkersMin,
 	}
-}
 
-type scaleClusterResults struct {
-	// number of workers according to our info, just before the PATCH call.
-	numWorkersBefore int
-	// number of workers to add, just before the PATCH call. Might be negative.
-	numWorkersToAdd int
-	// number of workers as of the PATCH call response.
-	numWorkersAfter int
+	if !cmd.Flags().Changed(cmdWorkersMinName) {
+		scaleArgs.workersMin = minBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) {
+		scaleArgs.workersMax = maxBefore
+	}
+	if !cmd.Flags().Changed(cmdWorkersMaxName) && !cmd.Flags().Changed(cmdWorkersMinName) && cmd.Flags().Changed("num-workers") {
```

Comment:
What about having a constant for `num-workers` as well?

Reply:
Done.